x/interchainqueries/keeper/keeper.go

```diff
@@ -195,9 +197,23 @@ func (k Keeper) GetQueryResultByID(ctx sdk.Context, id uint64) (*types.QueryResu
 	return &query, nil
 }
 
-func (k Keeper) removeQueryResultByID(ctx sdk.Context, id uint64) {
+// removeQueryResult removes the query result data. For a KV query it deletes the *types.QueryResult
+// stored by the query ID, for a TX query it deletes all tx hashes relative to the query.
+func (k Keeper) removeQueryResult(ctx sdk.Context, query *types.RegisteredQuery) {
 	store := ctx.KVStore(k.storeKey)
-	store.Delete(types.GetRegisteredQueryResultByIDKey(id))
+	queryType := types.InterchainQueryType(query.GetQueryType())
+	switch {
+	case queryType.IsKV():
+		store.Delete(types.GetRegisteredQueryResultByIDKey(query.Id))
+	case queryType.IsTX():
+		prefixStore := prefix.NewStore(ctx.KVStore(k.storeKey), types.GetSubmittedTransactionIDForQueryKeyPrefix(query.Id))
+		iterator := prefixStore.Iterator(nil, nil)
+		defer iterator.Close()
+		for ; iterator.Valid(); iterator.Next() {
+			txHashKey := iterator.Key()
+			prefixStore.Delete(txHashKey)
```

Comment:
Every delete operation costs 1000 gas, are you sure we do not meet out of gas error during iteration process?

Reply:
good catch. I also suspect that some queries with lots of recorded transactions are going to be unprofitable for the remover even given the deposit, so this will result in garbage queries that nobody wants to remove